### DISCUSSION TITLE
How to set up an Expo monorepo with two apps and a shared component library based on Tamagui?

### DISCUSSION THREAD

**Discussion by inakineitor** [tamagui/tamagui#120] [General]
I apologize in advance if this is not the place to ask this type of question. If that is the case, let me know and I will delete/close the issue.

Back to the question, I created a monorepo following [Expo's instruction](https://docs.expo.dev/guides/monorepos/) and it has been working fine. One of the apps is a blank expo app while the other one is halfway built, with Tamagui already configured. Now, I would like to create a common component package so that both apps in the monorepo can share the same component library (built on top of Tamagui).

Both apps share a design language, onboarding, authentication screens and a backend, so the more things I can de-duplicate inside "packages", the better.

Does anyone know how I would go about setting this up? I looked at the next-expo-solito started for guidance but was not able to come up with my own solution by just looking at how that repo is architected since my needs are somewhat different.

I am currently not using Lerna/Nx/Turno, but open to adding it to the monorepo if required (it will likely never have more than 2 or 3 apps).

Forgot to add: I am using EAS, if that makes any difference.

**Comment by natew** [comment]
I'd suggest modeling it after the starter repo, which shares a design system between a next.js app and expo, but you can really share it between any number of apps. `npx create-tamagui-app@latest`

**Comment by apperside** [comment]
If it can be of any help, [here](https://github.com/apperside/nx-tamagui-example) you can find a monorepo built with NX+(bare) react native+next js where the mobile and web app share a common UI library
